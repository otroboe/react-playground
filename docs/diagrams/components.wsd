@startuml
title Components

skinparam cloud {
  backgroundColor Skyblue
}

:User GPS: as User
cloud "GET /locations" as API_locations_all
cloud "GET /locations/:id" as API_locations_one
cloud "GET /locations/:id/shedules" as API_locations_one_schedule
cloud "GET /movies/:uuid" as API_movies_one

node App
node Root
node Header
node Main
node Home
node CinemaList
node CinemaListActions
node CinemaListItems
node CinemaPage
node CinemaPageDetails
node ScheduleList
node ScheduleListItems

App -- Root
Root -- Header
Root -- Main
Main -- Home
Main -- CinemaList
CinemaList -- CinemaListActions
CinemaList ~~ CinemaListItems
CinemaList ~~ CinemaListItems
Main -- CinemaPage
CinemaPage -- CinemaPageDetails
CinemaPage -- ScheduleList
ScheduleList ~~ ScheduleListItems
ScheduleList ~~ ScheduleListItems

CinemaList -up-> API_locations_all
CinemaListItems --> User
CinemaPage -up-> API_locations_one
ScheduleList -up-> API_locations_one_schedule
ScheduleList ~up~> API_movies_one
ScheduleList ~up~> API_movies_one

@enduml
