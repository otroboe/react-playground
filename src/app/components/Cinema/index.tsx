import React from 'react';
import { Route, Switch } from 'react-router-dom';
import CinemaList from './CinemaList';
import CinemaPage from './CinemaPage';

export class Cinema extends React.Component<{}, {}> {
  public render() {
    return (
      <Switch>
        <Route exact path="/cinema" component={CinemaList}/>
        <Route path="/cinema/:id" component={CinemaPage}/>
      </Switch>
    );
  }
}

export default Cinema;
