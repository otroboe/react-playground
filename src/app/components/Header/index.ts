export * from './Header';
export * from './MenuLanguage';
