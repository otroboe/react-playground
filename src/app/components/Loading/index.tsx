import React from 'react';
import { Spinner, Classes, Intent } from '@blueprintjs/core';

import './style.less';

interface LoadingProps {}

export class Loading extends React.Component<LoadingProps, {}> {
  public render() {
    return (
      <div className="loading">
        <Spinner className={Classes.LARGE} intent={Intent.NONE}/>
      </div>
    );
  }
}

export default Loading;
