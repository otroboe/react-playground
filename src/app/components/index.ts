export * from './Cinema';
export * from './Header';
export * from './Home';
export * from './Loading';
export * from './Map';
export * from './Schedule';
