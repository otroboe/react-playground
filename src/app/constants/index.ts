export * from './api';
export * from './geoloc';
export * from './locales';
export * from './settings';
export * from './stores';
