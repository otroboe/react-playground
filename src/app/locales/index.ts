import { translations as dataEn } from './en';
import { translations as dataFr } from './fr';

export const translations = {
  en: dataEn,
  fr: dataFr,
};

export * from './ILocale';
