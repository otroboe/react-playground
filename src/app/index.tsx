import React from 'react';
import { History } from 'history';
import { Router, Route, Switch } from 'react-router-dom';
import { inject } from 'mobx-react';

import { Root } from 'app/containers/Root';
import { Cinema, Home, MapPage } from 'app/components';
import { STORE_GEOLOC } from 'app/constants';
import { GeolocStore } from 'app/stores';

export interface AppProps {
  history: History;
}

@inject(STORE_GEOLOC)
export class App extends React.Component<AppProps, {}> {
  componentDidMount() {
    const geolocStore = this.props[STORE_GEOLOC] as GeolocStore;
    
    geolocStore.startWatch();
  }
  
  componentWillUnmount() {
    const geolocStore = this.props[STORE_GEOLOC] as GeolocStore;
    
    geolocStore.clearWatch();
  }

  render() {
    return (
      <Router history={this.props.history}>
        <Root>
          <Switch>
            <Route exact path="/" component={Home}/>
            <Route path="/cinema" component={Cinema}/>
            <Route path="/map" component={MapPage}/>
          </Switch>
        </Root>
      </Router>
    );
  }
}

export default App;
