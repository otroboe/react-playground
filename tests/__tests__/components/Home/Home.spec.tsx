import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { Home } from 'app/components';

describe('Home page', () => {
  test('Renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Welcome message', () => {
    const home = shallow(<Home />);
  
    expect(home.text()).toEqual('Welcome You !');
  });
});
