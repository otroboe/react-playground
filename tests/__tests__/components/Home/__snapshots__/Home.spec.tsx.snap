// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Renders correctly 1`] = `
<div
  className="home"
>
  <h1>
    Welcome You !
  </h1>
</div>
`;
